src/plugins/hiddenColumns/hiddenColumns.js

```diff
@@ -157,7 +157,7 @@ class HiddenColumns extends BasePlugin {
     let destinationHideConfig = currentHideConfig;
 
     if (isConfigValid) {
-      destinationHideConfig = currentHideConfig.filter(column => columns.includes(column) === false);
+      destinationHideConfig = Array.from(new Set(this.#hiddenColumnsMap.getNotHiddenIndexes().concat(columns)));
```

Comment:
The `Array.from(new Set(` pattern is repeating three times in this PR.

Shouldn't we move common code or use one of the array helpers? https://github.com/handsontable/handsontable/blob/560eb12677ec1b6b02d86b1f7d9715410f69d903/src/helpers/array.js#L246-L317


and optimize it there with `Array.from(new Set(` pattern so all use cases across HoT are optimized as well?

----

i.e.

```suggestion
      destinationHideConfig = arrayUnique(this.#hiddenColumnsMap.getNotHiddenIndexes().concat(columns));
```

or without `concat`:

```suggestion
      destinationHideConfig = getUnionOfArrays(this.#hiddenColumnsMap.getNotHiddenIndexes(), columns);
```

Reply:
We discussed about that some times ago. Maybe some extra issue should be done for that? It should be tested properly.

PS. My code has been changed. It won't be used anymore. However, a hint about code duplication may be still accurate. I will think whether some extra methods/helpers should be added.

Reply:
I'm not sure if parts of the current code should be moved to one place. I have no idea how I can extend `TrimmingMap` and `HidingMap` - whether some generic methods for getting **copy** of the state after `untrim`/`unhide` should be provided 🤔